src/ui.ts

```diff
@@ -162,6 +173,14 @@ const createMap = (container: string | HTMLElement, options?: any): Promise<any>
     unit: 'm',
   })
 
+  if (options.urlHash) {
+    parsedUrl = await urlMustBeParsed(getHashUrl())
+    const mapOptions = prepareMapOptions(parsedUrl)
```

Comment:
You can probably do something like that:
```suggestion
    options = prepareMapOptions(parsedUrl, options)
```
To  unnecessary forEach